### ISSUE TITLE
class_name and 'as' operator causes game crash if exported for release (debug export is ok)

### ISSUE THREAD

**Issue by meld-cp** [godotengine/godot#25124]
**Godot version:**
v3.1.beta2

**OS/device including version:**
Win10 Pro (1809)

**Issue description:**
Silent crash when run after a release export (32 and 64bit) if class_name and 'as' operator used  (see minimal project)

**Steps to reproduce:**
Write code like:
```gdscript
func _ready() -> void:
	# Below will silent crash when run after a release export (win.exe).
	# But will work fine in debug export or run through the editor
	# MySprite is defined via class_name
	var a = $MySprite as MySprite
	#var a = get_node("MySprite") as MySprite  # will also crash
	
	# But this works fine with release export
	var b = $MySprite
	if b is MySprite:
		print("ok")
```

**Minimal reproduction project:**
[ReleaseExportCrash.zip](https://github.com/godotengine/godot/files/2775289/ReleaseExportCrash.zip)

**Comment by bojidar-bg** [comment]
CC @vnen

**Comment by bojidar-bg** [comment]
@meld-cp Just to make sure: Are you using 3.1 beta **2** export templates, and not templates for an earlier version of Godot?

**Comment by meld-cp** [comment]
Correct, downloaded the beta and templates from the beta2 blog post posted recently.
 https://downloads.tuxfamily.org/godotengine/3.1/beta2/

**Comment by akien-mga** [comment]
Tested on 3.1 beta 7, it seems to work fine for me with both Linux 64-bit release and Windows 64-bit release (tested the latter via WINE).

Closing as fixed, please comment if you can still reproduce the issue and we'll reopen it.

**Comment by meld-cp** [comment]
Using 32bit version of Godot 3.1b7 on Win 10 Pro (64bit). 
- 32bit release of win exe works fine
- 64bit release won't start. (no splash-screen at all)

I also tried 64bit version of Godot 3.1**b6** and the release build worked there.

So maybe the issue is fixed but the b7 issue is related to #26243 ?  (I'm running old hardware, pre 2011)

I can try again when beta 8 is out.
